### PR TITLE
[SPARK-23063][K8S] K8s changes for publishing scripts (and a couple of other misses)

### PR DESCRIPTION
**Pull request by foxish**
## What changes were proposed in this pull request?

Including the `-Pkubernetes` flag in a few places it was missed.

## How was this patch tested?

checkstyle, mima through manual tests.

### CODE REVIEW

**Comment by foxish** [comment]
cc/ @felixcheung @sameeragarwal @vanzin @rxin

File: dev/create-release/releaseutils.py [hunk 1]
```diff
@@ -185,6 +185,7 @@ def get_commits(tag):
     "graphx": "GraphX",
     "input/output": CORE_COMPONENT,
     "java api": "Java API",
+    "kubernetes": "Kubernetes",
```
Comment (felixcheung) [inline dev/create-release/releaseutils.py]: this is for the PR title [foo] - I think [k8s] is more widely used, maybe both

Comment (foxish) [reply dev/create-release/releaseutils.py]: Can we supply a list/tuple there? I've updated it to `K8S`, but sometimes folks have written `k8s` or `kubernetes` in the PR titles by the looks of it.

Comment (felixcheung) [reply dev/create-release/releaseutils.py]: so looks like this is for both commit title and JIRA component field... which isn't quite perfect (for example, not R here)

but in any case, multiple left value can map to the same right value, it looks like

Comment (foxish) [reply dev/create-release/releaseutils.py]: Ah, okay, I misread that previously - updated the mapping, and it looks like the script turns things to lower case anyway, so `k8s` and `kubernetes` ought to cover everything. Thanks!

Comment (felixcheung) [reply dev/create-release/releaseutils.py]: yes, I think this looks right

**Comment by foxish** [comment]
Can someone trigger a retest please?

**Comment by sameeragarwal** [comment]
verified changes to the release scripts and a did simple grep check -- both LGTM. Thanks!

**Review by felixcheung — APPROVED** [review]

**Comment by felixcheung** [comment]
thanks! merged to master/2.3

**Pull request closed without merging**